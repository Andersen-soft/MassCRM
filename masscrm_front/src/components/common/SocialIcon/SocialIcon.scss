.icon {
  width: 24px;
  height: 24px;

  &_active {
    cursor: pointer;

    &:hover {
      opacity: 0.8;
    }
  }
}
