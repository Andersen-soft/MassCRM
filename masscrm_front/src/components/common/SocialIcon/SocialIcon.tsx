import React, { FC } from 'react';
import { styleNames } from 'src/services';
import { applicationPath } from 'src/constants';
import { socialNetworks } from 'src/data/socialNetworks';
import { Tooltip } from '@material-ui/core';
import { tooltipStyle } from 'src/styles/ToolTip.style';
import { ISocialIconProps } from './interfaces';
import styles from './SocialIcon.scss';

const sn = styleNames(styles);

export const SocialIcon: FC<ISocialIconProps> = ({ link, socialName }) => {
  const styleTooltip = tooltipStyle();
  const icon = (
    <use
      xlinkHref={`${applicationPath}assets/svg/sprite.svg#${
        socialNetworks.includes(socialName) ? socialName : 'angel'
      }`}
    />
  );
  const href = socialName === 'skype' && link ? `skype:${link}` : link;

  return href && link ? (
    <Tooltip title={link} placement='bottom' classes={styleTooltip}>
      <a href={href} rel='noreferrer' target='_blank'>
        <svg className={`${sn('icon')} ${!!link && sn('icon_active')}`}>
          {icon}
        </svg>
      </a>
    </Tooltip>
  ) : null;
};
