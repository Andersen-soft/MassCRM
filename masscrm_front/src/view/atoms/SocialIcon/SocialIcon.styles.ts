import { createStyles, makeStyles } from '@material-ui/core';

export const useStyles = makeStyles(() =>
  createStyles({
    icon: {
      width: '24px',
      height: '24px'
    },

    activeIcon: {
      cursor: 'pointer',
      '&:hover': {
        opacity: '0.8'
      }
    }
  })
);
