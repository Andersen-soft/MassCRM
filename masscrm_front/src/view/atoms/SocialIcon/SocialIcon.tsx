import React, { FC } from 'react';
import { Tooltip } from '@material-ui/core';
import {
  applicationPath,
  BLANK,
  BOTTOM,
  NO_REFERRER,
  socialNetworks
} from 'src/constants';
import { tooltipStyles } from 'src/styles';
import { useStyles } from './SocialIcon.styles';

interface IProps {
  link?: string;
  socialName: string;
}

export const SocialIcon: FC<IProps> = ({ link, socialName }) => {
  const styles = useStyles();
  const tooltipClasses = tooltipStyles();

  const icon = (
    <use
      xlinkHref={`${applicationPath}assets/svg/sprite.svg#${
        socialNetworks.includes(socialName) ? socialName : 'angel'
      }`}
    />
  );
  const href = socialName === 'skype' && link ? `skype:${link}` : link;

  return href && link ? (
    <Tooltip title={link} placement={BOTTOM} classes={tooltipClasses}>
      <a href={href} rel={NO_REFERRER} target={BLANK}>
        <svg className={`${styles.icon} ${!!link && styles.activeIcon}`}>
          {icon}
        </svg>
      </a>
    </Tooltip>
  ) : null;
};
