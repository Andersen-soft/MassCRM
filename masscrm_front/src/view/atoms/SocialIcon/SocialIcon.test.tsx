import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { SocialIcon } from './SocialIcon';

describe('SocialIcon', () => {
  const icon = shallow(<SocialIcon socialName='linkedin' />);

  it('SocialIcon should render correctly', () => {
    expect(shallowToJson(icon)).toMatchSnapshot();
  });
});
