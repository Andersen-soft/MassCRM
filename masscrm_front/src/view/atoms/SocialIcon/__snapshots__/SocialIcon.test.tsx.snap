// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialIcon SocialIcon should render correctly 1`] = `""`;
